hooks/application.go

```diff
@@ -43,7 +40,8 @@ func (m *applicationMutator) Handle(ctx context.Context, req admission.Request)
 	if err := m.dec.Decode(req, app); err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
-	if app.GetNamespace() == m.config.ArgoCD.Namespace {
+	// An application created with argocd cli will have an empty namespace.
+	if app.GetNamespace() == "" || app.GetNamespace() == m.config.ArgoCD.Namespace {
```

Comment:
`metadata.namespace` and `metadata.name` fields of the object passed to a mutating webhook can be undetermined.
The destination namespace is available from the `admission.Request`.